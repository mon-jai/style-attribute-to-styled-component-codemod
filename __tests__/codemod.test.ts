import { readFile } from "fs/promises"

import { expect, test } from "@jest/globals"
import jscodeshiftCore, { API } from "jscodeshift"

import transform from "../codemod.ts"

async function transformFile(filename: string) {
  const path = `__tests__/fixtures/${filename}`
  const source = await readFile(path, "utf-8")
  const jscodeshift = jscodeshiftCore.withParser("babel")

  return transform({ path, source }, { jscodeshift } as API, {})
}

test("components inheritance being parsed correctly", async () => {
  const transformedSource = await transformFile("inheritance.jsx")
  return expect(transformedSource).toMatchSnapshot()
})

test("unnamed components should be updated if appropriate", async () => {
  const transformedSource = await transformFile("unnamed-component.jsx")
  return expect(transformedSource).toMatchSnapshot()
})

test("user-named components should not be overwritten", async () => {
  const transformedSource = await transformFile("user-named-component.jsx")
  return expect(transformedSource).toMatchSnapshot()
})
