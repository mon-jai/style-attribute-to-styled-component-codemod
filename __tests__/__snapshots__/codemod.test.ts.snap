// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components inheritance being parsed correctly 1`] = `
"import styled from "styled-components"


const Div0 = styled.div\`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
\`

const Div1 = styled(Div0)\`
  flex-grow: 1;
\`

export default function Component() {
  return (
    (<Div1>
      <Div1 />
    </Div1>)
  );
}
"
`;

exports[`unnamed components should be updated if appropriate 1`] = `
"import styled from "styled-components"
const Div1 = styled.div\`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
\`


const Div0 = styled(Div1)\`
  flex-shrink: 1;
\`

const Div2 = styled(Div1)\`
  flex-grow: 1;
\`

export default function Component() {
  return (
    (<Div0>
      <Div2 />
    </Div0>)
  );
}
"
`;

exports[`user-named components should not be overwritten 1`] = `
"import styled from "styled-components"
const Div0 = styled.div\`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
  flex-grow: 1;
\`


const Foo = styled.div\`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  flex-direction: column;
  flex-shrink: 1;
\`

export default function Component() {
  return (
    (<Foo>
      <Div0 />
    </Foo>)
  );
}
"
`;
